src/main/java/gameframework/gui/GameStatusBar.java

```diff
@@ -12,39 +11,34 @@
 
 public class GameStatusBar implements Observer {
 
-	protected Label lifeText, scoreText;
-	protected Label lifeValue, scoreValue;
-
-	protected final ObservableValue<Integer> score;
-	protected final ObservableValue<Integer> life;
-
-	public GameStatusBar(ObservableValue<Integer> score,
-			ObservableValue<Integer> life) {
-		this.score = score;
-		this.life = life;
-		score.addObserver(this);
-		life.addObserver(this);
-	}
+	protected final ArrayList<GameStatusBarElement> elements = new ArrayList<GameStatusBarElement>();
 
 	public Container getContainer() {
-		JPanel c = new JPanel();
+		JPanel container = new JPanel();
 		GridBagLayout layout = new GridBagLayout();
-		c.setLayout(layout);
-		lifeText = new Label("Lives:");
-		scoreText = new Label("Score:");
-		lifeValue = new Label();
-		scoreValue = new Label();
-		c.add(lifeText);
-		c.add(lifeValue);
-		c.add(scoreText);
-		c.add(scoreValue);
+		container.setLayout(layout);
+
+		for (GameStatusBarElement element : elements) {
+			container.add(element.getElementText());
+			container.add(element.getElementValue());
+		}
+
 		update();
-		return c;
+		return container;
+	}
+
+	public void add(GameStatusBarElement newElement) {
+		this.elements.add(newElement);
+	}
+
+	public void addAll(List<GameStatusBarElement> newElements) {
+		this.elements.addAll(newElements);
 	}
 
 	public void update() {
-		lifeValue.setText(Integer.toString(life.getValue()));
-		scoreValue.setText(Integer.toString(score.getValue()));
+		for (GameStatusBarElement element : elements) {
+			element.getElementValue().setText(element.getString());
```

Comment:
please delegate this code to `GameStatusBarElement`. This will look like:

```java
public void update() {
    for (GameStatusBarElement element : elements) {
        element.update();
    }
}
```